test/integration/endpoints/endpoints_test.go

```diff
@@ -20,27 +21,65 @@ func TestMain(m *testing.M) {
 }
 
 func TestGoodEndpoints(t *testing.T) {
-	ns := TestHelper.GetLinkerdNamespace()
-	vizNs := TestHelper.GetVizNamespace()
+	ctx := context.Background()
+	controlNs := TestHelper.GetLinkerdNamespace()
 	testDataPath := "testdata"
 	cmd := []string{
 		"diagnostics",
 		"endpoints",
-		fmt.Sprintf("linkerd-dst.%s.svc.cluster.local:8086", ns),
-		fmt.Sprintf("grafana.%s.svc.cluster.local:3000", vizNs),
-		fmt.Sprintf("linkerd-identity.%s.svc.cluster.local:8080", ns),
-		fmt.Sprintf("linkerd-proxy-injector.%s.svc.cluster.local:443", ns),
-		fmt.Sprintf("tap.%s.svc.cluster.local:8088", vizNs),
-		fmt.Sprintf("web.%s.svc.cluster.local:8084", vizNs),
+		fmt.Sprintf("linkerd-dst.%s.svc.cluster.local:8086", controlNs),
+		fmt.Sprintf("linkerd-identity.%s.svc.cluster.local:8080", controlNs),
+		fmt.Sprintf("linkerd-proxy-injector.%s.svc.cluster.local:443", controlNs),
 	}
 
 	if !TestHelper.ExternalPrometheus() {
-		cmd = append(cmd, fmt.Sprintf("prometheus.%s.svc.cluster.local:9090", vizNs))
+		cmd = append(cmd, fmt.Sprintf("nginx.%s.svc.cluster.local:8080", "linkerd-endpoints-test"))
+		TestHelper.WithDataPlaneNamespace(ctx, "endpoints-test", map[string]string{}, t, func(t *testing.T, ns string) {
+			nginx, err := TestHelper.LinkerdRun("inject", "testdata/nginx.yaml")
+			if err != nil {
+				testutil.AnnotatedFatal(t, "unexpected error", err)
+			}
```

Comment:
I think you can just add the inject annotation into the yaml directly to avoid this step...

Reply:
Replaced with `TestHelper.Kubectl` so we avoid reading the file in (`TestHelper.KubectlApply` creates the deployment from stdin so we'd have to read the manifest in). Good suggestion 👍🏻!